test/server/cards/02_SHD/upgrades/Protector.spec.js

```diff
@@ -0,0 +1,23 @@
+describe('Protector', function() {
+    integration(function() {
+        describe('Protector\'s ability', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['battlefield-marine'],
+                    },
+                    player2: {
+                        groundArena: [{ card: 'snowspeeder', upgrades: ['protector'] }]
+                    }
+                });
+            });
+
+            it('should be sentinel', function () {
+                this.player1.clickCard(this.battlefieldMarine);
+                //Snowspeeder automatically selected due to sentinel
```

Comment:
We need to confirm that the attack actually resolves (confirm that the damage happened and / or cards are moved to discard), there are a lot of ways that the attack resolution could silently fail and move directly to player 2 being active